packages/ckeditor5-clipboard/src/clipboardmarkersutils.ts

```diff
@@ -262,76 +385,121 @@ export default class ClipboardMarkersUtils extends Plugin {
+			const position = fakeMarker.markerElement && writer.createPositionBefore( fakeMarker.markerElement );
+			let prevFakeMarker: FakeMarkerRangeConstruct | null = acc[ fakeMarker.name ];
+
+			// Handle scenario when tables clone cells with the same fake node. Example:
+			//
+			// <cell><fake-marker-a></cell> <cell><fake-marker-a></cell> <cell><fake-marker-a></cell>
+			//                                          ^ cloned                    ^ cloned
+			//
+			// The easiest way to bypass this issue is to rename already existing in map nodes and
+			// set them new unique name.
+			if ( prevFakeMarker && prevFakeMarker.start && prevFakeMarker.end ) {
+				acc[ this._getUniqueMarkerName( fakeMarker.name ) ] = acc[ fakeMarker.name ];
+				prevFakeMarker = null;
+			}
 
-				// [ foo <fake-marker>aaa</fake-marker> test ]
-				//                    ^
-				// Handle case when marker is between start and end of selection.
-				const startPosition = writer.createPositionAt( startElement, 'before' );
-				writer.remove( startElement );
+			acc[ fakeMarker.name ] = {
+				...prevFakeMarker!,
+				[ fakeMarker.type ]: position
+			};
 
-				const endPosition = writer.createPositionAt( endElement, 'before' );
-				writer.remove( endElement );
+			if ( fakeMarker.markerElement ) {
+				writer.remove( fakeMarker.markerElement );
+			}
 
-				acc[ markerName ] = new Range( startPosition, endPosition );
+			return acc;
+		}, {} );
 
-				return acc;
-			}, {} );
+		return mapValues(
+			fakeMarkersRanges,
+			range => new Range(
+				range.start || writer.createPositionFromPath( rootElement, [ 0 ] ),
+				range.end || writer.createPositionAt( rootElement, 'end' )
+			)
+		);
 	}
 
 	/**
-	 * Returns object that contains mapping between marker names and corresponding `$marker` elements.
+	 * Returns array that contains list of fake markers with corresponding `$marker` elements.
 	 *
 	 * For each marker, there can be two `$marker` elements or only one (if the document fragment contained
 	 * only the beginning or only the end of a marker).
 	 *
 	 * @param writer An instance of the model writer.
-	 * @param element The element to be checked.
+	 * @param rootElement The element to be checked.
 	 */
-	private _getAllFakeMarkersFromElement( writer: Writer, element: Element | DocumentFragment ): Record<string, Array<Element>> {
-		return Array
-			.from( writer.createRangeIn( element ) )
-			.reduce<Record<string, Array<Element>>>( ( fakeMarkerElements, { item } ) => {
-				if ( item.is( 'element', '$marker' ) ) {
-					const fakeMarkerName = item.getAttribute( 'data-name' ) as string;
+	private _getAllFakeMarkersFromElement( writer: Writer, rootElement: Element | DocumentFragment ): Array<FakeMarker> {
+		const foundFakeMarkers = Array
+			.from( writer.createRangeIn( rootElement ) )
+			.flatMap( ( { item } ): Array<FakeMarker> => {
+				if ( !item.is( 'element', '$marker' ) ) {
+					return [];
+				}
```

Comment:
I dislike using functions for hacky purposes (even if it is a popular use).

Also, don't you rather want to iterate by items in the range, not tree walker values? Right now you are getting each element twice.

Finally, I don't think you need to put the type (maybe it was needed for `flatMap()` but IDK if it is needed for `map()` in this case:

```suggestion
			.from( writer.createRangeIn( rootElement ).getItems() )
			.filter( item => item.is( 'element', '$marker' ) )
			.map( item => {
```

WDYT?

Reply:
@scofalik 

>  I dislike using functions for hacky purposes (even if it is a popular use). Also, don't you rather want to iterate by items in the range, not tree walker values? Right now you are getting each element twice.

Me too. Probably it has to be refactored in next steps. For now - it works besides lots of weird edge cases.

> Finally, I don't think you need to put the type (maybe it was needed for flatMap() but IDK if it is needed for map() in this case:

It's not necessary, but it's faster to find a place when already mapped array type mismatches the shape of `FakeMarker`. Instead of looking from bottom to top of the array mapping chain, you see directly one `map` method which maps wrong value. Small thing but sometimes kinda useful.